src/js/components/app.test.js

```diff
@@ -58,16 +61,25 @@ describe('App Component', () => {
   });
 
   it('works as intended', async () => {
-    window.localStorage.getItem.mockReturnValueOnce('false');
-    Object.defineProperty(window.document, 'cookie', {
-      writable: true,
-      value: 'JWT=omnomnom'
+    const store = mockStore({
+      ...state,
+      users: {
+        ...state.users,
+        currentUser: 'notNull'
+      }
     });
-    render(
+    window.localStorage.getItem.mockReturnValueOnce('false');
+    const ui = (
       <Provider store={store}>
         <App />
       </Provider>
     );
-    await act(async () => jest.advanceTimersByTime(900500));
+    const { rerender } = render(ui);
+    await act(async () => jest.advanceTimersByTime(timeout + 500));
+    cookies.get.mockReturnValue('');
+    await waitFor(() => rerender(ui));
+    await waitFor(() => expect(screen.queryByText(/Version:/i)).not.toBeInTheDocument());
+    expect(screen.queryByText(/Northern.tech/i)).toBeInTheDocument();
+    expect(screen.queryByText(`© ${mockDate.getFullYear()} Northern.tech AS`)).toBeInTheDocument();
```

Comment:
just for my peace of mind, `testDate` from `mockData` is set as the current date of the test run, right?

Reply:
Yes, that date is used to mock the environment/ browser date, so what the footer renders is a `new Date().getFullYear()` and these should be equal - unless the mock doesn't work... and keeping it at a live date would break many other parts of the test that either show a date we need to compare or use the date as a seed for a request e.g.

Reply:
not to mention runs right before midnight on Ner Years Eve.
nice.